costmodel/costmodel.go

```diff
@@ -44,52 +42,20 @@ const (
 )
 
 type CostModel struct {
-	Controller *Controller
-}
+	Cache ClusterCache
 
-func NewCostModel(podListWatcher cache.ListerWatcher) *CostModel {
-	cm := &CostModel{}
-	cm.ContainerWatcher(podListWatcher)
-	return cm
+	stop chan struct{}
 }
 
-func (cm *CostModel) ContainerWatcher(podListWatcher cache.ListerWatcher) {
-
-	// create the workqueue
-	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+func NewCostModel(client kubernetes.Interface) *CostModel {
```

Comment:
The underlying controllers should provide a HasSynced() function for an "initial sync" of your underlying caches. I'd prefer if this waited to return until all the controllers have synced. No need to do anything fancy, I think a while loop in KubernetesClusterCache should do the trick.